test/src/test_utils.c

```diff
@@ -125,19 +126,34 @@ ts_debug_shippable_fatal_after_n_rows(PG_FUNCTION_ARGS)
 
 /*
  * Broken send/receive functions for int4 that throw after an (arbitrarily
- * chosen prime) number of rows.
- * Use ERROR, not FATAL, because PG versions < 14 are unable to report a FATAL
- * error to the access node before closing the connection, so the test results
- * would be different.
+ * chosen prime or configured) number of rows.
  */
-#define ARBITRARY_PRIME_NUMBER 7103
+static void
+broken_sendrecv_throw()
+{
+	int throw_after = 7103; /* an arbitrary prime */
```

Comment:
nit: just a thought, that could also use random number generator with seed to have a broader coverage  and deterministic output

Reply:
Yep, could be nice, although we don't have a way to set random seed for the entire test run now. And we'll have to mask the particular numbers from the output somehow.